frontend/src/store/modules/eceweApp.js

```diff
@@ -1,153 +1,137 @@
+        console.info(`Failed to get ECEWE Application - ${error}`);
         throw error;
       }
     },
-    async saveECEWE({ state, rootState, dispatch }) {
-      if (!localStorage.getItem('jwtToken')) { // DONT Call api if there is no token.
-        console.log('unable to save because you are not logged in');
-        throw new Error('unable to save ecewe application because you are not logged in');
-      }
+    async saveECEWE({ state, commit }) {
+      checkSession();
       try {
-        // Ensure the state of ECEWE questions is accurate.
-        const updatedState = {
-          belongsToUnion: (state.optInECEWE==0)?null:state.belongsToUnion,
-          fundingModel: (state.belongsToUnion==0 || state.belongsToUnion == null)?null:state.fundingModel,
-          confirmation: (state.belongsToUnion==0 || state.belongsToUnion == null)?null:state.confirmation,
-        };
-        Object.assign(state, updatedState);
-
-        // Flag if user is opting out of ECEWE, but has previously saved ECEWE facilities with
-        // opt-in/out values. We will check this flag after saving the parent ECEWE record below. 
-        let updateFacilities = false;
-        if (state.facilities != null || state.facilities?.length > 0) {
-          state.facilities.find(facility => {
-            if (facility.eceweApplicationId != null && facility.optInOrOut != null) {
-              facility.optInOrOut = null;
-              updateFacilities = true;
-            }
-          });
+        console.info('ECEWEApp.saveECEWE.state.eceweModel = ');
+        console.info(JSON.parse(JSON.stringify(state.eceweModel)));
+        console.info('ECEWEApp.saveECEWE.state.loadedModel = ');
+        console.info(JSON.parse(JSON.stringify(state.loadedModel)));
+        if (isEqual(state.eceweModel, state.loadedModel)) {
+          console.info('ECEWEApp.saveECEWE: no model changes (NOT saving).');
+          return;
+        } else {
+          console.info('ECEWEApp.saveECEWE: model changes (SAVING).');
         }
-        let payload = JSON.parse(JSON.stringify(state));
-        
-        // remove attributes we are not updating before sending payload.
+        let payload = { ...state.eceweModel };
+        payload = JSON.parse(JSON.stringify(payload));
         delete payload.facilities;
         // Save ECEWE parent record.
-        let response = await ApiService.apiAxios.patch(ApiRoutes.APPLICATION_ECEWE + '/' + rootState.organization.applicationId, payload);
-
-        // If parent ECEWE save was successfull and ECEWE facilties have been flagged for update, issue update on ECEWE facilties as their
-        // previous optinout responses are no longer applicable.
-        if (response?.status == '200' && updateFacilities) {
-          dispatch('saveECEWEFacilities');
-        }
+        let response = await ApiService.apiAxios.patch(ApiRoutes.APPLICATION_ECEWE + '/' + state.applicationId, payload);
+        commit('setLoadedModel', {...state.eceweModel});
```

Comment:
move this to be before the save.  If the save fails, then the childcare provider will never be able to leave the page.